### ISSUE TITLE
Check does not error as expected when missing stanza

### ISSUE THREAD

**Issue by dwsteele** [pgbackrest/pgbackrest#281]
I intentionally installed backrest only on the db server and when I call check without a stanza it appears to succeed, but with a stanza it fails as expected.
```
postgres@localhost:~$ pgbackrest --stanza=<stanza> check
ERROR [124]: protocol version mismatch: PGBACKRESTREMOTE 1.08
postgres@localhost:~$ pgbackrest check
postgres@localhost:~$
```
The new version was `1.10`.

**Comment by dwsteele** [comment]
Also observed by Adrian in issue #282:

Somewhere between 1.06 and 1.09 the behavior of pgbackrest changed when a wrong (not defined) stanza is used.

1.06:
pgbackrest --stanza=asdf --log-level-console=info backup; echo $?
ERROR [112]: backup command requires option: db-path
HINT: does this stanza exist?
112

1.09:
pgbackrest --stanza=asdf --log-level-console=info backup; echo $?
112

The current release also contains this bug 1.11:
pgbackrest --stanza=asdf --log-level-console=info backup; echo $?
112

**Comment by dwsteele** [comment]
Fixed in d18cd7ad0faa80c4ceacd708d0ae1e964f170a7a, @disco-stu.  It looks like this was introduced in 1.09, though I did not bisect to determine for sure.  In any case, the new code should be more robust.
